### ISSUE TITLE
Migrate from CSV Doc Update 

### ISSUE THREAD

**Issue by robert-s-lee** [cockroachdb/docs#5341]
Re: [Migrate from CSV](https://cockroachlabs.com/docs//v19.1/migrate-from-csv.html)

## Issue Description

Below is not correct:

If one of the following characters appears in a field, the field must be enclosed by double quotes:
`double quote (")`

## Suggested Resolution 

- Rows 1 and 2 do not require leading double quote
- Row 3 double quote is ok as well
- Row 4 required ending quote because it started with a quote

```
   1|***TESTDATA ABC DEF “A” GHI10|***COPY TEST ABC DEF “A” GHI10
   2|***TESTDATA ABC DEF "A" GHI10|***COPY TEST ABC DEF "A" GHI10
   3|"***TESTDATA ABC DEF "A" GHI10"|***COPY TEST ABC DEF "A" GHI10
   4|""***TESTDATA ABC DEF "A" GHI10"|***COPY TEST ABC DEF "A" GHI10
```

```
create table q1 (
  id1 int primary key,
  id2 string,
  id3 string
);

root@:26257/defaultdb> import table q1 create using 'nodelocal:///quotetestdata.ddl'
   csv data ('nodelocal:///quotetestdata.csv')
   with delimiter = '|';
        job_id       |  status   | fraction_completed | rows | index_entries | system_records | bytes
+--------------------+-----------+--------------------+------+---------------+----------------+-------+
  481823351026843649 | succeeded |                  1 |    4 |             0 |              0 |   297
(1 row)

Time: 88.438ms

root@:26257/defaultdb> select * from q1;
  id1 |              id2               |              id3
+-----+--------------------------------+--------------------------------+
    1 | ***TESTDATA ABC DEF “A” GHI10  | ***COPY TEST ABC DEF “A” GHI10
    2 | ***TESTDATA ABC DEF "A" GHI10  | ***COPY TEST ABC DEF "A" GHI10
    3 | ***TESTDATA ABC DEF "A" GHI10  | ***COPY TEST ABC DEF "A" GHI10
    4 | "***TESTDATA ABC DEF "A" GHI10 | ***COPY TEST ABC DEF "A" GHI10
(4 rows)
```

**Comment by lhirata** [comment]
@rolandcrosby @robert-s-lee does this mean that if double quotes appear in a field, they do not need to be enclosed by double quotes? I tried to replicate what Robert did but I get an error, so I'm confused:

```> import table q1 (
  id1 int primary key,
  id2 string,
  id3 string
) CSV DATA ('nodelocal:///quotestestdata.csv');
pq: nodelocal:///quotestestdata.csv: row 1: reading CSV record: parse error on line 1, column 21: bare " in non-quoted-field```

Doesn't this error imply you need the field to be enclosed by double quotes?

**Comment by spaskob** [comment]
Can you try using the import mode 'DELIMITED` instead.
So
```
import table q1 (
  id1 int primary key,
  id2 string,
  id3 string
) DELIMITED DATA ('nodelocal:///quotestestdata.csv');
```

**Comment by spaskob** [comment]
More context in this related issue: https://github.com/cockroachdb/cockroach/issues/39820

**Comment by robert-s-lee** [comment]
@lhirata the command is missing `with delimiter = '|';` to use the supplied file.  @spaskob what does `DELIMITED DATA` take as the delimiter?  Can `|` be specified?

**Comment by spaskob** [comment]
yes, try appending `WITH fields_escaped_by ='|';`
You can also take a look here: https://github.com/cockroachdb/cockroach/issues/39820#issuecomment-528082881

**Comment by robert-s-lee** [comment]
Is there a beta or alpha build available with `DELIMITED DATA`?

**Comment by spaskob** [comment]
I'm not familiar with the release process, but this is the PR that introduced it: https://github.com/cockroachdb/cockroach/pull/40482

**Comment by rolandcrosby** [comment]
@robert-s-lee I don't believe this has landed in any of the alpha or beta builds yet, but you can try the latest bleeding-edge build to test this functionality: https://github.com/cockroachdb/cockroach/commit/0568c6332f45689f953b8d6e7f6d2b067b385c6d

(Note that this build is otherwise unsuitable for production use, as there are still some beta-blocking bugs included, but you can at least try the DELIMITED format.)

The options to DELIMITED are as follows:

* `rows_terminated_by`: the character to indicate new lines in the input file (defaults to `\n`)
* `fields_terminated_by`: the delimiter used to separate fields in each input line
* `fields_enclosed_by`: the character that can optionally enclose fields
* `fields_escaped_by`: the character that, when preceding one of the above characters, causes that character to be interpreted literally

I was able to import the above snippet and preserve all quotes using the following statement:

```
IMPORT TABLE test (a string, b string, c string) DELIMITED DATA ('nodelocal:///test.dsv') WITH fields_terminated_by='|'
```

That produced the following table data:

```
   a   |                b                 |               c
+------+----------------------------------+--------------------------------+
     1 | ***TESTDATA ABC DEF “A” GHI10    | ***COPY TEST ABC DEF “A” GHI10
     2 | ***TESTDATA ABC DEF "A" GHI10    | ***COPY TEST ABC DEF "A" GHI10
     3 | "***TESTDATA ABC DEF "A" GHI10"  | ***COPY TEST ABC DEF "A" GHI10
     4 | ""***TESTDATA ABC DEF "A" GHI10" | ***COPY TEST ABC DEF "A" GHI10
```

Unfortunately, I can't figure out a combination of options (with our current syntax) that could import this data while also stripping the quotes at the beginning and end of column B of rows 3 and 4. The SQL `btrim` function could potentially be used here, except that it operates recursively, so it would strip both quotes from the beginning of row 4 column B.

**Comment by spaskob** [comment]
Thanks for the detailed explanation Roland!
Isn't

   - fields_enclosed_by: the character that can optionally enclose fields

boing to help with the outer quotes?

**Comment by rolandcrosby** [comment]
Unfortunately not - I tried using `with fields_terminated_by='|', fields_enclosed_by='"'` and got an `unmatched field enclosure` error on row 4.

**Comment by spaskob** [comment]
Interesting, this code has not been exercised much, it's quite possible
this is a bug; will look into this as well.

**Comment by robert-s-lee** [comment]
@rolandcrosby @spaskob we changed behavior with 19.2 beta.  

- The import command below worked in 19.1, but with 19.2 beta `import csv` errors out.  

```
import table q1 create using 'nodelocal:///quotetestdata.ddl'
   csv data ('nodelocal:///quotetestdata.csv')
   with delimiter = '|';
pq: nodelocal:///quotetestdata.csv: row 2: reading CSV record: parse error on line 2, column 22: bare " in non-quoted-field
```

- `import  DELIMITED` completes but the `"` is not stripped:

```
import table q1 create using 'nodelocal:///quotetestdata.ddl'
   DELIMITED data ('nodelocal:///quotetestdata.csv')
 WITH fields_terminated_by='|';

> select * from q1;
  id1 |               id2                |              id3
+-----+----------------------------------+--------------------------------+
    1 | ***TESTDATA ABC DEF “A” GHI10    | ***COPY TEST ABC DEF “A” GHI10
    2 | ***TESTDATA ABC DEF "A" GHI10    | ***COPY TEST ABC DEF "A" GHI10
    3 | "***TESTDATA ABC DEF "A" GHI10"  | ***COPY TEST ABC DEF "A" GHI10
    4 | ""***TESTDATA ABC DEF "A" GHI10" | ***COPY TEST ABC DEF "A" GHI10
```

- `fields_enclosed_by='"'` is required to bring back the original `import csv` behavior 
```
import table q1 create using 'nodelocal:///quotetestdata.ddl'
   DELIMITED data ('nodelocal:///quotetestdata.csv')
 WITH fields_terminated_by='|',fields_enclosed_by='"' ;

> select * from q1;
  id1 |              id2               |              id3
+-----+--------------------------------+--------------------------------+
    1 | ***TESTDATA ABC DEF “A” GHI10  | ***COPY TEST ABC DEF “A” GHI10
    2 | ***TESTDATA ABC DEF "A" GHI10  | ***COPY TEST ABC DEF "A" GHI10
    3 | ***TESTDATA ABC DEF "A" GHI10  | ***COPY TEST ABC DEF "A" GHI10
    4 | "***TESTDATA ABC DEF "A" GHI10 | ***COPY TEST ABC DEF "A" GHI10

```

**Comment by jseldess** [comment]
@lnhsingh, can you provide an update here? Do you have the info you need to make docs changes and resolve the issue?

**Comment by lnhsingh** [comment]
This issue should've been closed by #5857. Closing now.
